sycl/include/CL/sycl/accessor.hpp

```diff
@@ -248,16 +248,23 @@ template <typename DataT, int Dimensions, access::mode AccessMode,
 class image_accessor
 #ifndef __SYCL_DEVICE_ONLY__
     : public detail::AccessorBaseHost {
-  size_t MImageSize;
   size_t MImageCount;
+  size_t MImageSize;
 #else
 {
-  /*
-  // TODO: Define the datatype here based on Dimensions, AccessMode and
-  AccessTarget.
-  __ocl_image[Dim]d[array/non_array]_[AM]_t MImage;
-  __init(__ocl_imagexx_t Image) { MImage = Image; }
-  */
+
+  using OCLImageTy = typename detail::opencl_image_type<Dimensions, AccessMode,
+                                                        AccessTarget>::type;
+  OCLImageTy MImageObj;
+  char MPadding[(sizeof(detail::AccessorBaseHost) +
+                 sizeof(size_t /*MImageSize*/) +
+                 sizeof(size_t /*MImageCount*/)) -
+                sizeof(OCLImageTy)];
+
+public:
+  void image_accessor_init(OCLImageTy Image) { MImageObj = Image; }
```

Comment:
Minor. Please, rename to imageAccessorInit then.

Reply:
done.